edweiss-app/__test__/showtime/LectureScreen.test.tsx

```diff
@@ -0,0 +1,180 @@
+
+// Mock dependencies
+jest.mock('expo-router', () => ({
+    useLocalSearchParams: jest.fn(),
+}));
+
+jest.mock('@/hooks/firebase/firestore', () => ({
+    usePrefetchedDynamicDoc: jest.fn(),
+}));
+
+jest.mock('@/config/firebase', () => ({
+    CollectionOf: jest.fn(),
+    getDownloadURL: jest.fn(),
+}));
+
+jest.mock('@/config/i18config', () => ({
+    __esModule: true,
+    default: jest.fn((key) => key),
+}));
+
+// Updated mock for expo-screen-orientation
+jest.mock('expo-screen-orientation', () => ({
+    lockAsync: jest.fn(),
+    unlockAsync: jest.fn(),
+    OrientationLock: { LANDSCAPE: 'LANDSCAPE', PORTRAIT_UP: 'PORTRAIT_UP' },
+    addOrientationChangeListener: jest.fn((callback) => {
+        callback({ orientationInfo: 'mockOrientation' });
+        return {
+            remove: jest.fn(),
+        };
+    }),
+    removeOrientationChangeListener: jest.fn(),
+}));
+
+jest.mock('@react-navigation/native-stack', () => {
+    const actualNav = jest.requireActual('@react-navigation/native-stack');
+    return {
+        ...actualNav,
+        createNativeStackNavigator: jest.fn(() => ({
+            Navigator: ({ children }: { children: ReactNode; }) => <>{children}</>,
+            Screen: ({ children }: { children: ReactNode; }) => <>{children}</>,
+        })),
+    };
+});
+
+// Mock the RouteHeader component if it's using Stack.Screen
+jest.mock('@/components/core/header/RouteHeader', () => {
+    const MockRouteHeader: FC<{ children: ReactNode; }> = ({ children }) => <>{children}</>;
+    return MockRouteHeader;
+});
+
+// Mock react-native-pdf to prevent native calls
+jest.mock('react-native-pdf', () => {
+    return () => <div data-testid="mock-pdf-viewer">PDF Viewer Mock</div>;
+});
+
+// Mock react-native-blob-util to prevent NativeEventEmitter errors
+jest.mock('react-native-blob-util', () => ({
+    fs: {
+        dirs: {
+            DocumentDir: 'mockDocumentDir',
+        },
+    },
+    config: jest.fn(() => ({
+        fetch: jest.fn(),
+    })),
+}));
+
+// Mock Firebase Messaging to avoid NativeEventEmitter errors
+jest.mock('@react-native-firebase/messaging', () => ({
+    __esModule: true,
+    default: jest.fn(() => ({
+        onMessage: jest.fn(),
+        onNotificationOpenedApp: jest.fn(),
+        getInitialNotification: jest.fn(),
+    })),
+}));
+
+// Mock Data
+const mockLectureData = {
+    data: {
+        pdfUri: 'mock-pdf-uri',
+        audioTranscript: ['Transcript for Page 1', 'Transcript for Page 2'],
+        questions: [
+            { id: '1', question: 'Question 1' },
+            { id: '2', question: 'Question 2' },
+        ],
+    },
+};
+
+describe('LectureScreen Component', () => {
+    beforeEach(() => {
+        (useLocalSearchParams as jest.Mock).mockReturnValue({
+            courseNameString: 'mockCourse',
+            lectureIdString: 'mockLectureId',
+        });
+        (usePrefetchedDynamicDoc as jest.Mock).mockReturnValue([mockLectureData]);
```

Comment:
I just want to know why are you using an array of `mockLectureData` as an argument to `usePrefetchedDynamicDoc`. In my opinion, it might be better to directly return the `mockLectureData` (not wrapped in an array), assuming the real hook returns data directly rather than within an array.

Here's a fix that may work:

```
(usePrefetchedDynamicDoc as jest.Mock).mockReturnValue(mockLectureData);
```

Reply:
thank you very much this will be useful, I have tried many things with my tests thats why some things may appear a bit weird sorry

Reply:
Yes I know, I am also testing my remote control and faced the same issue as you. However, as you may have noticed my other remark below, for the moment we can stay with it.